lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
@markusweimer @anupam128  I calculated some time stats. With 50 million dimensional vector and IMRU, the  average GC time is around 100 millisecond, the communication latency is around 250 milli seconds while actual data communication time is close to 10 seconds. I think it really sort of matters for real-time applications with real strict time requirements.